cve_bin_tool/input_engine.py

```diff
@@ -115,24 +115,17 @@ def strip_remark(detail) -> str:
                         ref = affect["ref"]
                         version = None
                         vendor = None
+                        # Expected ref format is urn:cbt:{bom_version}/{vendor}#{product}-{version}
                         if "#" in ref:
                             # Extract product information after # delimiter
                             p = ref.split("#")[1]
                             # Last element is version, rest is product which may include -
                             elements = p.rsplit("-", 1)
                             product = elements[0]
                             version = elements[1]
-                            # Now find vendor
-                            vendor_package_pair = self.cvedb.get_vendor_product_pairs(
-                                product
-                            )
-                            if vendor_package_pair != []:
-                                # FIXME https://github.com/intel/cve-bin-tool/issues/2320
-                                vendor = vendor_package_pair[
-                                    len(vendor_package_pair) - 1
-                                ]["vendor"]
-                            else:
-                                vendor = "UNKNOWN"
+                            # Extract vendor between / and #
+                            vendor = ref.split("/")[1].split("#")[0]
```

Comment:
I think we probably need to do a little pre-validation so this doesn't break if the string is malformed.  e.g.:

```python
ref = "blah"
vendor = ref.split("/")[1].split("#")[0]
```

gives...

```
Traceback (most recent call last):
  File "testme.py", line 4, in <module>
    vendor = ref.split("/")[1].split("#")[0]
IndexError: list index out of range
```

so maybe...
```suggestion
                            tempstring = ref.split("/")
                            if (len(tempstring) > 1):
                                vendor = ref.split("/")[1].split("#")[0]
                            else:
                                vendor = "UNKNOWN"
```

There's probably some more elegant way but that at least avoids a traceback.